api/class-wc-rest-dev-mailchimp-settings-controller.php

```diff
@@ -0,0 +1,307 @@
+		$handler                            = MailChimp_Woocommerce_Admin::connect();
+		$data                               = $handler->validate( $parameters );
+
+		update_option( 'mailchimp-woocommerce', $data );
+
+		return rest_ensure_response( $data );
+	}
+
+
+	/**
+	 * Get current newsletter settings
+	 *
+	 * @param WP_REST_Request,
+	 * @return WP_REST_Response, list of newsletters
+	 *
+	 */
+	public function get_newsletter_settings( $request ) {
+		$handler = MailChimp_Woocommerce_Admin::connect();
+		$data    = $handler->getMailChimpLists();
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Update newsletter settings
+	 *
+	 * @param WP_REST_Request, newsletter settings
+	 * @return WP_REST_Response, updated MailChimp settings
+	 *
+	 */
+	public function update_newsletter_settings( $request ) {
+		$parameters                         = $request->get_params();
+		$parameters['mailchimp_active_tab'] = 'newsletter_settings';
+		$handler                            = MailChimp_Woocommerce_Admin::connect();
+		$options                            = get_option( 'mailchimp-woocommerce', array() );
+		$mailchimp_active_tab               = $options['active_tab'];
+	  $data                               = $handler->validate( $parameters );
+
+		// if previous active tab was sync then we still want sync
+		// because this call is just an update and not part of setup
+		if( 'sync' === $mailchimp_active_tab  ) {
+			$data['mailchimp_active_tab'] = 'sync';
+		}
+
+		update_option( 'mailchimp-woocommerce', $data);
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Synchronization status between MailChimp plugin and Mailchimp
+	 *
+	 * @param WP_REST_Request,
+	 * @return WP_REST_Response, synchronization status
+	 *
+	 */
+	public function get_sync_status( $request ) {
+		$handler                  = MailChimp_Woocommerce_Admin::connect();
+		$mailchimp_total_products = $mailchimp_total_orders = 0;
+		$store_id                 = mailchimp_get_store_id();
+		$product_count            = mailchimp_get_product_count();
+		$order_count              = mailchimp_get_order_count();
+		$store_syncing            = false;
+		$last_updated_time        = get_option( 'mailchimp-woocommerce-resource-last-updated' );
+		$account_name             = 'n/a';
+		$mailchimp_list_name      = 'n/a';
+
+		if (!empty($last_updated_time)) {
+		    $last_updated_time = mailchimp_date_local( $last_updated_time );
+		}
+
+		if ( ( $mailchimp_api = mailchimp_get_api() ) && ( $store = $mailchimp_api->getStore( $store_id ) ) ) {
+
+		    $store_syncing = $store->isSyncing();
+
+		    if ( ( $account_details = $handler->getAccountDetails() ) ) {
+		        $account_name = $account_details['account_name'];
+		    }
+
+		    try {
+		        $products = $mailchimp_api->products( $store_id, 1, 1 );
+		        $mailchimp_total_products = $products['total_items'];
+		        if ( $mailchimp_total_products > $product_count ) $mailchimp_total_products = $product_count;
```

Comment:
This line still needs braces around it.

`if ( $mailchimp_total_products > $product_count ) $mailchimp_total_products = $product_count;`

to

```
if ( $mailchimp_total_products > $product_count ) {
    $mailchimp_total_products = $product_count;
}
```

The same thing goes for line 268 for orders.